apps/arweave/src/ar_data_sync.erl

```diff
@@ -2429,30 +2442,34 @@ validate_proof2(Args) ->
 	end.
 
 validate_data_path(DataRoot, Offset, TXSize, DataPath, Chunk) ->
-	ValidatePathResult =
-		case ar_merkle:validate_path_strict_data_split(DataRoot, Offset, TXSize, DataPath) of
-			false ->
-				case ar_merkle:validate_path_strict_borders(DataRoot, Offset, TXSize,
-						DataPath) of
-					false ->
-						false;
-					R ->
-						{false, R}
-				end;
-			R ->
-				{true, R}
+	Base = ar_merkle:validate_path_strict_borders(DataRoot, Offset, TXSize, DataPath),
+	Strict = ar_merkle:validate_path_strict_data_split(DataRoot, Offset, TXSize, DataPath),
+	Rebase = ar_merkle:validate_path2(DataRoot, Offset, TXSize, DataPath),
+	Result =
+		case {Base, Strict, Rebase} of
+			{false, false, false} ->
+				false;
+			{_, {_, _, _} = R, _} ->
```

Comment:
What do you think about renaming R to better indicate which of the validation results it comes from? e.g.
```
		case {Base, Strict, Rebase} of
			{false, false, false} ->
				false;
			{_, {_, _, _} = StrictResult, _} ->
				StrictResult;
			{_, _, {_, _, _} = RebaseResult} ->
				RebaseResult;
			{{_, _, _} = BaseResult, _, _} ->
				BaseResult
		end,
```